overlay/lower/usr/share/udhcpc/default.script

```diff
@@ -74,7 +74,7 @@ case "$1" in
 			echo "nameserver $i" >>$tmpfile
 		done
 		mv $tmpfile $realconf
-		if [ -n "$ntpsrv" ] && [ -w /tmp/ntp.conf ]; then
+		if [ -n "$ntpsrv" ] && { [ ! -f /tmp/ntp.conf ] || stat -c%a /tmp/ntp.conf | grep -q '[67]'; }; then
```

Comment:
We can simplify that clusterfuck of conditions. /tmp/ntp.conf should always be in place to that moment. We should create it first copying from /etc/default/ntp.conf, unconditionally. Thst is our safe ground. So checking for that file should always return true and can be eliminated from the conditions. In that case, you only need to check if there is ntp data in dhcp response and there is, can you override the working copy with that data.

Reply:
we need a field on the ntp form allowing the user to lock the config to the preset values ignoring the dhcp supplied values (opt-out mode), or explicitly allowing dhcp to override these values (opt-in mode). then, depending on the mode we chose, the file from /etc/default/ is not only being copied but also `chmod` to 444. 

then, in the script above, you check the mode for matching 444 and bail out if it is. not for absense of 6 and 7 but for a concrete match, as that is the only mode being set by the script.

Reply:
@themactep 
Agreed - I like the idea of having the behavior configurable for the user via the UI.  Would NTP from DHCP be opt-in or opt-out by default?  
For now, user NTP config from the UI takes precedence over DHCP.
Can we merge this change and then I'll make another change to support the ability to change modes?